.github/workflows/ci.yaml

```diff
@@ -653,3 +640,25 @@ jobs:
           [ "${{matrix.settings.name}}" = "Sqlserver" ] && exit 0
 
           just slt-bin --protocol=flightsql --exclude '*/tunnels/ssh' ${{matrix.settings.path}}
+
+  docker-push:
+    name: Build and Push Docker Image
+    if: github.event_name == 'push' && github.repository == 'GlareDB/glaredb' && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest-8-cores
```

Comment:
depending on if we decide to reuse the previous artifacts or not we can downside this.

Reply:
I'd be happy to accomplish both. The code as-is was just copy-pasta'd from the release variant, but since this runs on every push to `main` and is generally OK to take longer, I'm happy to downsize.